tests/unit_tests/test_subtensor.py

```diff
@@ -1440,11 +1453,52 @@ def test_do_serve_axon_is_success(subtensor, mocker):
     assert result == (True, None)
 
 
+def test_do_serve_axon_tls_is_success(subtensor, mocker):
+    """Successful do_serve_axon call."""
+    # Prep
+    fake_wallet = mocker.MagicMock()
+    fake_call_params = get_fake_call_params()
+    fake_call_params["certificate"] = Certificate("fake_cert")
+    fake_wait_for_inclusion = True
+    fake_wait_for_finalization = True
+
+    subtensor.substrate.submit_extrinsic.return_value.is_success = True
+
+    # Call
+    result = subtensor._do_serve_axon(
+        wallet=fake_wallet,
+        call_params=fake_call_params,
+        wait_for_inclusion=fake_wait_for_inclusion,
+        wait_for_finalization=fake_wait_for_finalization,
+    )
+
+    # Asserts
+    subtensor.substrate.compose_call.assert_called_once_with(
+        call_module="SubtensorModule",
+        call_function="serve_axon_tls",
+        call_params=fake_call_params,
+    )
+
+    subtensor.substrate.create_signed_extrinsic.assert_called_once_with(
+        call=subtensor.substrate.compose_call.return_value,
+        keypair=fake_wallet.hotkey,
+    )
+
+    subtensor.substrate.submit_extrinsic.assert_called_once_with(
+        subtensor.substrate.create_signed_extrinsic.return_value,
+        wait_for_inclusion=fake_wait_for_inclusion,
+        wait_for_finalization=fake_wait_for_finalization,
+    )
+
+    subtensor.substrate.submit_extrinsic.return_value.process_events.assert_called_once()
+    assert result == (True, None)
```

Comment:
```suggestion
    assert result[0] is True
    assert result[1] is None
```

It's important to do comparisons like this because a result of `(1, None)` would also pass this assert.